app/policies/user_policy.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class UserPolicy < ApplicationPolicy
+  def index?
+    user_admin?
+  end
+
+  def show?
+    user_admin?
+  end
+
+  def update?
+    user_admin?
+  end
+
+  def edit?
+    user_admin? || current_user.id == model.id
+  end
```

Comment:
so finally, the user is allowed to edit himself, but is disallowed to save changes, right?

Reply:
fix this